src/game/Game.java

```diff
@@ -107,98 +103,164 @@ public int getPlayerTurn() {
 		return playerTurn;
 	}
 
+	public Owner getOwner() {
+		return Owner.values()[(getPlayerTurn() - 1)];
+	}
+
+	public Owner getOtherOwner() {
+		if (getPlayerTurn() == 1) {
+			return Owner.values()[1];
+		} else {
+			return Owner.values()[0];
+		}
+	}
+
 	public void setPlayerTurn(int playerTurn) {
 		this.playerTurn = playerTurn;
 	}
 
-	@Deprecated
-	public AbstractPiece getSpace(int row, int column) {
-		return this.currentBoard.getPieceAt(new Coordinate(row, column));
+	public AbstractPiece getPieceAt(Coordinate coor) {
+		return this.currentBoard.getPieceAt(coor);
 	}
-	
+
 	// refactor for future pull request
 	public boolean checkCoor(int row, int column) {
 		return this.checkCoor(new Coordinate(row, column));
 	}
-	
+
 	public boolean checkCoor(Coordinate coor) {
 		return this.currentBoard.pieceAt(coor);
 	}
 
 	/**
 	 * 
-	 * @param row
-	 * @param column
-	 * @param dir
-	 *            0: up, 1: right, 2: down, 3: left
-	 * @return returns true if the move made successfully, otherwise returns false
+	 * @param moveToMake
+	 * @return
 	 */
-	public boolean move(int row, int column, int dir) {
-		MoveCommand moveToMake;
-		if (!isValidMoveFromSquare(row, column))
-			return false;
-		switch (dir) {
-		case 0:
-			// Moving UP
-			moveToMake = new MoveUp(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		case 1:
-			// Moving RIGHT
-			moveToMake = new MoveRight(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		case 2:
-			// Moving DOWN
-			moveToMake = new MoveDown(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		case 3:
-			// Moving LEFT
-			moveToMake = new MoveLeft(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		default:
+	public boolean move(BoardState boardState, Coordinate start, Coordinate dest) {
+		AbstractPiece piece = this.currentBoard.getPieceAt(start);
+		if (isFrozen(start)) {
 			return false;
 		}
+		if ((piece instanceof Rabbit)) {
+			if (((piece.getOwner() == Owner.values()[0]) && (dest.equals(start.up())))
+					|| ((piece.getOwner() == Owner.values()[1]) && (dest.equals(start.down())))) {
+				// Cannot move a Rabbit backwards unless it has been dragged
+				return false;
+			}
+		}
+		if (start.isOrthogonallyAdjacentTo(dest)
+				&& this.getBoardState().getPieceAt(start).getOwner() == this.getOwner()) {
+			RegularMove moveToMake = new RegularMove(boardState, start, dest);
+			this.currentBoard = moveToMake.execute();
+			this.moves.add(moveToMake);
+			endMove();
+			return true;
+		}
+		return false;
+	}
```

Comment:
Handles how pieces are allowed to move. These two branches cover Rabbits (pawns) not being allowed to move backwards, and general move logic.